lib/kcerts/ssh.go

```diff
@@ -268,8 +205,101 @@ func SignPublicKey(ca ssh.Signer, certType uint32, principals []string, ttl time
 		ValidPrincipals: principals,
 		Permissions:     ssh.Permissions{},
 	}
-	if err := cert.SignCert(rand.Reader, ca); err != nil {
+	s, err := p.Signer()
+	if err != nil {
+		return nil, err
+	}
+	if err := cert.SignCert(rand.Reader, s); err != nil {
 		return nil, err
 	}
 	return cert, nil
 }
+
+func OpenSSHEncode21559PrivateKey(key ed25519.PrivateKey) ([]byte, error) {
+	b := openSSHMarshalED25519PrivateKey(key)
+	return pem.EncodeToMemory(&pem.Block{Type: "OPENSSH PRIVATE KEY", Bytes: b}), nil
+
+}
+
+// Note: copited from https://github.com/mikesmitty/edkey/blob/master/edkey.go
```

Comment:
we can't just copy in this code, see LICENSE here: https://github.com/mikesmitty/edkey/blob/master/LICENSE, as a bare minimum, we should put it in a directory, and have the LICENSE file in it.

Better: can we import the library like any other go dependency? WORKSPACE and such.